src/tap/data/ResultSetTableIterator.java

```diff
@@ -731,7 +731,7 @@ else if (colValue instanceof Boolean)
 			else if (colType != null && colValue != null && colType.type == DBDatatype.CHAR && (colType.length == 1 || colType.length <= 0) && colValue instanceof String)
 				colValue = ((String)colValue).charAt(0);
 			// if the column value is a geometrical object, it must be serialized in STC-S:
-			else if (translator != null && colType != null && colType.isGeometry()){
+			else if ((translator != null && colType != null && colType.isGeometry()) || colValue instanceof byte[]) {
```

Comment:
This was due to an incomplete initialisation of the `ResultSetTableIterator` and of the missing implementation of `SQLServerTranslator.translateGeometryFromDB(...)`.

I have fix the constructor of `ResultSetTableIterator` in order to use the automatic datatypes. According to some tests I ran, it should now work. Could you fetch - again :-( - the last commit and try?